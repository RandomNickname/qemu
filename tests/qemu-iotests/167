#! /bin/bash
#
# qemu-img dd test for the status option
#
# Copyright (C) 2016 Reda Sallahi
#
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; either version 2 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#

owner=fullmanet@gmail.com

seq="$(basename $0)"
echo "QA output created by $seq"

here="$PWD"
status=1

_cleanup()
{
    _cleanup_test_img
    rm -f "$TEST_IMG.out" "$TEST_IMG.out.stat" "$TEST_IMG.out.dd" \
          "$TEST_IMG.out.dd.stat"
}
trap "_cleanup; exit \$status" 0 1 2 3 15

. ./common.rc
. ./common.filter
. ./common.pattern

_supported_fmt raw
_supported_proto file
_supported_os Linux

echo
echo "== Creating image =="

size=1M
_make_test_img $size
_check_test_img

$QEMU_IO -c "write -P 0xa 149k 512k" "$TEST_IMG" | _filter_qemu_io

echo
echo "== Converting the image with dd status=noxfer =="

$QEMU_IMG dd if="$TEST_IMG" of="$TEST_IMG.out" conv=notrunc -O "$IMGFMT" \
          seek=10 skip=4 status=noxfer 2> "$TEST_IMG.out.stat"
TEST_IMG="$TEST_IMG.out" _check_test_img

dd if="$TEST_IMG" of="$TEST_IMG.out.dd" conv=notrunc status=noxfer \
   seek=10 skip=4 2> "$TEST_IMG.out.dd.stat"

echo
echo "== Compare the images with qemu-img compare =="

$QEMU_IMG compare "$TEST_IMG.out" "$TEST_IMG.out.dd"

echo
echo "== Compare the stat output =="

diff "$TEST_IMG.out.dd.stat" "$TEST_IMG.out.stat"

echo
echo "*** done"
rm -f "$seq.full"
status=0
