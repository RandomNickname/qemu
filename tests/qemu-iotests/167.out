QA output created by 167

== Creating image ==
Formatting 'TEST_DIR/t.IMGFMT', fmt=IMGFMT size=1048576
No errors were found on the image.
wrote 524288/524288 bytes at offset 152576
512 KiB, X ops; XX:XX:XX.X (XXX YYY/sec and XXX ops/sec)

== Converting the image with dd status=noxfer ==
No errors were found on the image.

== Compare the images with qemu-img compare ==
Images are identical.

== Compare the stat output ==

*** done
